<launch>
    <!-- Args that can be set from command line -->
    <arg name="marker_id" default="0"/>
    <arg name="agent_id" default="0"/>
    <arg name="poll_rate" default="10.0"/>
    <arg name="prediction_horizon" default="3.0"/>

    <!-- Camera settings -->
    <include file="matbot/launch/run_cam.launch"/>

    <!-- AR tag tracker -->
    <include file="lab4_cam/launch/ar_track.launch">
        <arg name="marker_size" value="16.5"/>
        <arg name="max_new_marker_error" value="0.05"/>
        <arg name="max_track_error" value="0.05"/>
        <arg name="cam_image_topic" value="/usb_cam/image_raw"/>
        <arg name="cam_info_topic" value="/usb_cam/camera_info"/>
        <arg name="output_frame" value="usb_cam"/>
    </include>

    <!-- AR State Collector -->
    <node name="ar_state_collector" pkg="matbot" type="ar_state_collector.py" output="screen">
        <param name="poll_rate" value="10.0" />
        <param name="history_length" value="20" />
        <rosparam param="ar_tag_ids">[0, 1, 2]</rosparam>
    </node>

    <!-- AR-Trajectron Bridge -->
    <node name="ar_trajectron_bridge" pkg="matbot" type="ar_trajectron_bridge.py" output="screen">
        <rosparam param="ar_tag_ids">[0, 1, 2]</rosparam>
        <param name="history_length" value="20" />
    </node>

    <!-- Trajectron Predictor -->
    <node name="trajectron_predictor" pkg="matbot" type="trajectron_predictor.py" output="screen">
        <param name="model_path" value="$(find matbot)/pretrained_models/model_registrar-12.pt" />
        <param name="config_path" value="$(find matbot)/pretrained_models/config.json" />
        <param name="log_dir" value="$(find matbot)/logs/trajectron" />
        <param name="prediction_horizon" value="3.0" />
        <rosparam param="ar_tag_ids">[0, 1, 2]</rosparam>
    </node>

    <!-- RViz for visualization -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find matbot)/rviz/ar_trajectron.rviz" />

</launch>